ledger-tool/src/program.rs

```diff
@@ -131,18 +131,50 @@ fn load_blockstore(ledger_path: &Path, arg_matches: &ArgMatches<'_>) -> Arc<Bank
     bank
 }
 
-pub trait RunSubCommand {
-    fn run_subcommand(self) -> Self;
+pub trait ProgramSubCommand {
+    fn program_subcommand(self) -> Self;
 }
 
-impl RunSubCommand for App<'_, '_> {
-    fn run_subcommand(self) -> Self {
+impl ProgramSubCommand for App<'_, '_> {
+    fn program_subcommand(self) -> Self {
+        let program_arg = Arg::with_name("PROGRAM")
+            .help(
+                "Program file to use. This is either an ELF shared-object file to be executed, \
+                 or an assembly file to be assembled and executed.",
+            )
+            .required(true)
+            .index(1);
+        let max_genesis_arg = Arg::with_name("max_genesis_archive_unpacked_size")
+            .long("max-genesis-archive-unpacked-size")
+            .value_name("NUMBER")
+            .takes_value(true)
+            .default_value("10485760")
+            .help("maximum total uncompressed size of unpacked genesis archive");
         self.subcommand(
+            SubCommand::with_name("program")
+        .about("Run to test, debug, and analyze on-chain programs.")
```

Comment:
One more request, can you please add:
```
.setting(AppSettings::InferSubcommands)
.setting(AppSettings::SubcommandRequiredElseHelp)
```
These are just both kind of convenience things:
- The first one allows something like `solana-ledger-tool program d` to yield `disassemble`
- The second one will throw error earlier if no subcommand specified

Reply:
done